lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs

```diff
@@ -254,6 +261,14 @@ spec = describe "SHARED_TRANSACTIONS" $ do
             , expectResponseCode HTTP.status202
             ]
 
+        let txid = getFromResponse (#id) submittedTx
+        let queryTx = Link.getTransaction @'Shared wal (ApiTxId txid)
+        rGetTx <- request @(ApiTransaction n) ctx queryTx Default Empty
+        verify rGetTx
+            [ expectResponseCode HTTP.status200
+            , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
+            ]
```

Comment:
unfortunately, sending tx is quite costly in integration testing and I try to avoid overtimes in CI. And in this case getTx or listTx is very natural way of checking the state of tx cycle. Hence, I utilized already present test. Possibly, Piotr will add additional dedicated tests. I just wanted to demonstrate it works

Reply:
I believe that more testing is required indeed. Both endpoints come with query parameters (same as their counterparts in Shelley scheme) so we need to make sure the behavior there is the same. There is a subsuite in Shelley transactions that checks that (https://github.com/input-output-hk/cardano-wallet/blob/master/lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs#L1373-L1898) so either we copy it and adapt, or perhaps the same can be tested on the unit level?

On my end I will first focus on adding appropriate tests on the e2e level building on top of recent changes introduced when testing tx history in https://github.com/input-output-hk/cardano-wallet/pull/3564.

Reply:
thanks Piotr, let's adapt them then.